subgraphs/badger-dao/src/handlers/deposit.ts

```diff
@@ -0,0 +1,44 @@
+import { Address, BigInt, ethereum } from "@graphprotocol/graph-ts";
+import { Vault } from "../../generated/schema";
+import { getOrCreateToken } from "../entities/Token";
+import { getOrCreateDeposit } from "../entities/Transaction";
+import { getPriceOfStakedTokens, getUsdPriceOfToken } from "./price";
+
+export function deposit(call: ethereum.Call, vault: Vault, amount: BigInt): void {
+  let deposit = getOrCreateDeposit(call.transaction.hash, call.transaction.index);
+
+  let totalSupply = vault.outputTokenSupply;
+  let balance = vault.inputTokenBalances[0];
+
+  let shares = totalSupply.isZero() ? amount : amount.times(totalSupply).div(balance);
```

Comment:
I think there is some logical issue in how you are calculating the shares minted. There are some logs in your subgraph where the deposit is negative which is wrong.

![image](https://user-images.githubusercontent.com/47497931/161353133-10b7d00d-15ab-4894-8720-cfe53e48b90d.png)

Reply:
> Hello @AP-Atul, I have gone through your subgraph and the `withdraws` and `deposits` entities are not properly linked to the `vault` entity. The value of `totalValueLockedUSD` is negative in some entities which I think is due to the fact that some of your deposits and withdrawal amounts are negative.
> 
> This is one of the problematic responses that I got when I queried your subgraph.
> 
> ```
> {
>    "id": "0x2b5455aac8d64c14786c3a29858e43b5945819c0",
>    "inputTokens": [
>      {
>        "id": "0x62b9c7356a2dc64a1969e19c23e4f579f9810aa7",
>        "name": "Convex CRV",
>        "symbol": "cvxCRV",
>        "decimals": 18
>      }
>    ],
>    "outputToken": {
>      "id": "0x2b5455aac8d64c14786c3a29858e43b5945819c0",
>      "name": "Badger Sett Convex CRV",
>      "symbol": "bcvxCRV",
>      "decimals": 18
>    },
>    "totalValueLockedUSD": "32242646.716810195533700310401018",
>    "totalVolumeUSD": "-44892730.549127507658901940165573",
>    "inputTokenBalances": ["11601476944924442195057071"],
>    "outputTokenSupply": "-403774488479015730829158",
>    "outputTokenPriceUSD": "0",
>    "createdTimestamp": "1624373395",
>    "createdBlockNumber": "12684635",
>    "name": "Badger Sett Convex CRV",
>    "symbol": "bcvxCRV",
>    "deposits": [],
>    "withdraws": []
> }
> ```

I have linked the withdraw and deposit entities.

Reply:
And for negative amount, trying to mitigate with checks